libeos-payg/tests/clock-jump-source.c

```diff
@@ -0,0 +1,151 @@
+/* -*- mode: C; c-file-style: "gnu"; indent-tabs-mode: nil; -*-
+ *
+ * Copyright © 2019 Endless Mobile, Inc.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at https://mozilla.org/MPL/2.0/.
+ *
+ * Alternatively, the contents of this file may be used under the terms of the
+ * GNU Lesser General Public License Version 2.1 or later (the "LGPL"), in
+ * which case the provisions of the LGPL are applicable instead of those above.
+ * If you wish to allow use of your version of this file only under the terms
+ * of the LGPL, and not to allow others to use your version of this file under
+ * the terms of the MPL, indicate your decision by deleting the provisions
+ * above and replace them with the notice and other provisions required by the
+ * LGPL. If you do not delete the provisions above, a recipient may use your
+ * version of this file under the terms of either the MPL or the LGPL.
+ */
+
+#include "config.h"
+
+#include <glib.h>
+#include <libeos-payg/clock-jump-source.h>
+#include <locale.h>
+#include <unistd.h>
+
+/* 60 FPS should be enough for anyone. */
+#define ITERATIONS 30
+#define USEC_PER_MSEC 1000
+
+static gboolean
+timeout_cb (gpointer data)
+{
+  guint *called = (guint *) data;
+  *called += 1;
+  return G_SOURCE_CONTINUE;
+}
+
+/* Test the timeout firing once */
```

Comment:
This test doesn't test a timeout firing, and it tests the source calling its callback twice, not once. I think given this tests both directions, and the same source firing more than once, the `many` test is redundant.